<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 6 Enrichment analyses | Omics Data Analysis" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2020-10-29" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML-full" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 6 Enrichment analyses | Omics Data Analysis</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a id="active-page" href="sec-gsea.html"><span class="toc-section-number">6</span> Enrichment analyses</a><ul class="toc-sections">
<li class="toc"><a href="#introduction-1"> Introduction</a></li>
<li class="toc"><a href="#gene-sets"> Gene sets</a></li>
<li class="toc"><a href="#over-representation-analysis-ora"> Over representation analysis (ORA)</a></li>
<li class="toc"><a href="#gene-set-enrichment-analysis-gsea"> Gene set enrichment analysis (GSEA)</a></li>
<li class="toc"><a href="#using-the-clusterprofiler-package"> Using the <code>clusterProfiler</code> package</a></li>
<li class="toc"><a href="#visualisation-of-enrichment-analyses"> Visualisation of enrichment analyses</a></li>
<li class="toc"><a href="#discussion"> Discussion</a></li>
</ul>
<a href="sec-ms.html"><span class="toc-section-number">7</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">8</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">9</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">10</span> Annex</a>
<a href="sec-si.html"><span class="toc-section-number">11</span> Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec:gsea" class="section level1">
<h1>
<span class="header-section-number">Chapter 6</span> Enrichment analyses</h1>
<p><strong>Motivation</strong>: at the end of this chapter, the students will be able to address the following points:</p>
<ul>
<li>What are the next steps after a differential expression analysis?</li>
<li>Why are enrichment analyses useful?</li>
<li>Understanding over representation analyses.</li>
<li>Understanding a gene set enrichment analysis.</li>
<li>Application of the <em><a href="https://bioconductor.org/packages/3.11/clusterProfiler">clusterProfiler</a></em> package.</li>
</ul>
<div id="introduction-1" class="section level2">
<h2>
<span class="header-section-number">6.1</span> Introduction</h2>
<ul>
<li><p>Differential expression analysis is <strong>univariate</strong> - each gene is tested on its own. This probably doesn't reflect the underlying biology - genes work in conjunction, not in isolation. One wouldn't expect that the effect of a drug, or a mutation, ... would lead to the perturbation of a single gene expression.</p></li>
<li><p>The univariate approach expects (or tests for) significant changes in single genes. Moderate effects in many (related) genes cannot, by definition be identified as statistically significant, even it such an effect may be biologically more plausible that one of a few large ones.</p></li>
<li><p>The goal of an enrichment analysis is to test for a group of related genes, called <strong>gene sets</strong>, and test whether the genes within these sets are enriched for differentially expression.</p></li>
</ul>
</div>
<div id="gene-sets" class="section level2">
<h2>
<span class="header-section-number">6.2</span> Gene sets</h2>
<p>While nothing would stop a user to create their own gene sets, the sets that are generally used stem from community-maintained resources.</p>
<div id="gene-ontology-go" class="section level3">
<h3>
<span class="header-section-number">6.2.1</span> Gene Ontology (GO)</h3>
<p>The <a href="http://geneontology.org/">Gene Ontology</a> <span class="citation">(Ashburner et al. <label for="tufte-mn-11" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-11" class="margin-toggle">2000<span class="marginnote">Ashburner, M, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, et al. 2000. “Gene Ontology: Tool for the Unification of Biology. the Gene Ontology Consortium.” <em>Nat Genet</em> 25 (1): 25–29. doi:<a href="https://doi.org/10.1038/75556">10.1038/75556</a>.</span>)</span> defines GO terms. These terms are based on a controlled vocabulary (to resuse the same words in a consistent way) and relations<label for="tufte-sn-13" class="margin-toggle sidenote-number">13</label><input type="checkbox" id="tufte-sn-13" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">13</span> Example of relations between terms are, for example, <em>is_a</em> (mitosis <em>is_a</em> cell cycle phase) or <em>part_of</em> (mitosis <em>part_of</em> M phase of mitotic cell cycle).</span> that define the directed links between terms. These relations define a hierarchy between GO terms: all term can be represented as a (direct acyclic) graph (DAG). There exist thus very general and very specific terms.</p>
<p>These terms are classed into three categories, called <em>namespaces</em>:</p>
<ul>
<li>Molecular Function (MF): molecular activities of gene products</li>
<li>Cellular Component (CC): where gene products are active</li>
<li>Biological Process (BP): pathways and larger processes made up of the activities of multiple gene products</li>
</ul>
<p>Here's an example of GO term <code>GO:0007155</code>, that describes cell adhesion.</p>
<pre><code>A Term from the GO ontology: GO:0007155
 Label: cell adhesion
  The attachment of a cell, either to another cell or to an underlying
  substrate such as the extracellular matrix, via cell adhesion
  molecules.</code></pre>
<p>If you look at the GO term <code>GO:0007155</code> <a href="http://amigo.geneontology.org/amigo/term/GO:0007155">entry on the Gene Ontology page</a>, you can find more details and, if you scroll down, example genes that are annotated with that term.</p>
<div class="figure" style="text-align: center">
<span id="fig:unnamed-chunk-113"></span>
<p class="caption marginnote shownote">
Figure 6.1: Gene ontology entry for the GO term for cell adhesion.
</p>
<img src="figs/go-screenshot.png" alt="Gene ontology entry for the GO term for cell adhesion." width="100%">
</div>
<p>The whole Gene Ontology is can be accessed in R with the <em><a href="https://bioconductor.org/packages/3.11/GO.db">GO.db</a></em> package.</p>
<p>In the code chunk below, we query the <code>GO.db</code> package through the <code>org.Hs.eg.db</code> interface. This <em>org.db</em>-type of packages for Homo sapiens enables to perform various queries for human genes, such as retrieving all gene symbols and ENTREZ identifiers (the <code>columns</code> below) that are annotated with a GO term (the <code>keys</code> below) of interest.</p>
<p>The GO term of interest here is focal adhesion:</p>
<pre><code>A cell-substrate junction that anchors the cell to the extracellular
matrix and that forms a point of termination of actin filaments. In
insects focal adhesion has also been referred to as hemi-adherens
junction (HAJ).</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"GO.db"</span>)
<span class="kw">library</span>(<span class="st">"org.Hs.eg.db"</span>)
GO_<span class="dv">0005925</span> &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,
                                    <span class="dt">keys =</span> <span class="st">"GO:0005925"</span>,
                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>),
                                    <span class="dt">keytype =</span> <span class="st">"GO"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(ENTREZID))</code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GO_<span class="dv">0005925</span></code></pre></div>
<pre><code>## # A tibble: 405 x 5
##    GO         EVIDENCE ONTOLOGY ENTREZID SYMBOL
##    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
##  1 GO:0005925 HDA      CC       60       ACTB  
##  2 GO:0005925 HDA      CC       70       ACTC1 
##  3 GO:0005925 ISS      CC       71       ACTG1 
##  4 GO:0005925 HDA      CC       81       ACTN4 
##  5 GO:0005925 HDA      CC       87       ACTN1 
##  6 GO:0005925 IMP      CC       88       ACTN2 
##  7 GO:0005925 IMP      CC       89       ACTN3 
##  8 GO:0005925 HDA      CC       102      ADAM10
##  9 GO:0005925 HDA      CC       118      ADD1  
## 10 GO:0005925 HDA      CC       214      ALCAM 
## # … with 395 more rows</code></pre>
<p>We have 405 genes matching this GO term. There are thus 405 genes in the GO_0005925 GO set.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the code above to extract the genes annotated with the <code>GO:0005813</code> term for the centrosome:</p>
<pre><code>A structure comprised of a core structure (in most organisms, a pair
of centrioles) and peripheral material from which a
microtubule-based structure, such as a spindle apparatus, is
organized. Centrosomes occur close to the nucleus during interphase
in many eukaryotic cells, though in animal cells it changes
continually during the cell-division cycle.</code></pre>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-16" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-16', 'sol-start-16')"></span>
</p>
<div id="sol-body-16" class="solution-body" style="display: none;">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GO_<span class="dv">0005813</span> &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db,
                     <span class="dt">keys =</span> <span class="st">"GO:0005813"</span>,
                     <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>),
                     <span class="dt">keytype =</span> <span class="st">"GO"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(ENTREZID))</code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GO_<span class="dv">0005813</span></code></pre></div>
<pre><code>## # A tibble: 483 x 5
##    GO         EVIDENCE ONTOLOGY ENTREZID SYMBOL
##    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
##  1 GO:0005813 IDA      CC       324      APC   
##  2 GO:0005813 IDA      CC       328      APEX1 
##  3 GO:0005813 IDA      CC       402      ARL2  
##  4 GO:0005813 IDA      CC       403      ARL3  
##  5 GO:0005813 IDA      CC       468      ATF4  
##  6 GO:0005813 IDA      CC       576      ADGRB2
##  7 GO:0005813 IBA      CC       582      BBS1  
##  8 GO:0005813 IDA      CC       585      BBS4  
##  9 GO:0005813 IDA      CC       598      BCL2L1
## 10 GO:0005813 IMP      CC       636      BICD1 
## # … with 473 more rows</code></pre>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="kyoto-encyclopedia-of-genes-and-genomes-kegg" class="section level3">
<h3>
<span class="header-section-number">6.2.2</span> Kyoto Encyclopedia of Genes and Genomes (KEGG)</h3>
<p><a href="https://www.genome.jp/kegg/pathway.html">KEGG</a> pathway is a collection of manually drawn and curated pathway maps representing current knowledge of the molecular interaction, reaction and relation networks.</p>
<p>The figure below shows the <a href="https://www.genome.jp/kegg-bin/show_pathway?hsa04110">pathways for the cell cycle</a> in humans.</p>
<div class="figure" style="text-align: center">
<span id="fig:unnamed-chunk-114"></span>
<p class="caption marginnote shownote">
Figure 6.2: KEGG pathway for cell cycle.
</p>
<img src="figs/hsa04110.png" alt="KEGG pathway for cell cycle." width="100%">
</div>
<p>The <em><a href="https://bioconductor.org/packages/3.11/KEGGREST">KEGGREST</a></em> package provides a client interface to the KEGG server.</p>
</div>
<div id="reactome" class="section level3">
<h3>
<span class="header-section-number">6.2.3</span> Reactome</h3>
<p>Alike KEGG patway, <a href="https://reactome.org/">Reactome</a> is a free, open-source, curated and peer-reviewed pathway database. The Bioconductor <em><a href="https://bioconductor.org/packages/3.11/reactome.db">reactome.db</a></em> package provides access to reactome maps and annotations within R.</p>
</div>
<div id="molecular-signatures-database-msigdb" class="section level3">
<h3>
<span class="header-section-number">6.2.4</span> Molecular Signatures Database (MSigDB)</h3>
<p>MSigDB is a collection of annotated gene sets for use with GSEA software. The MSigDB gene sets are divided into 9 collections:</p>
<ul>
<li>Hallmark gene sets (H) are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</li>
<li>Positional gene sets (C1) for each human chromosome and cytogenetic band.</li>
<li>Curated gene sets (C2) from online pathway databases, publications in PubMed, and knowledge of domain experts.</li>
<li>Regulatory target gene sets (C3) based on gene target predictions for microRNA seed sequences and predicted transcription factor binding sites.</li>
<li>Computational gene sets (C4) defined by mining large collections of cancer-oriented microarray data.</li>
<li>Ontology gene sets (C5) consist of genes annotated by the same ontology term.</li>
<li>Oncogenic signature gene (C6) sets defined directly from microarray gene expression data from cancer gene perturbations.</li>
<li>Immunologic signature gene sets (C7) defined directly from microarray gene expression data from immunologic studies.</li>
<li>Cell type signature gene sets (C8) curated from cluster markers identified in single-cell sequencing studies of human tissue.</li>
</ul>
<p>The <em><a href="https://CRAN.R-project.org/package=msigdbr">msigdbr</a></em> CRAN package provides the MSigDB gene sets in a standard R data frame with key-value pairs.</p>
</div>
<div id="input-data" class="section level3">
<h3>
<span class="header-section-number">6.2.5</span> Input data</h3>
<p>To illustrate enrichment analyses, we will use the DESeq2 results stored in the <code>res_tbl</code> variable, computed in the previous chapter.</p>
<p>We will focus on the genes that have an adjusted p-value (those that have been tested) and that have unique ENTREZ gene identifiers.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_tbl &lt;-<span class="st"> </span>res_tbl <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ENTREZID),
           <span class="op">!</span><span class="kw">is.na</span>(padj),
           <span class="op">!</span><span class="kw">duplicated</span>(ENTREZID)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">ENTREZID =</span> <span class="kw">as.character</span>(ENTREZID))

res_tbl</code></pre></div>
<pre><code>## # A tibble: 13,719 x 9
##    ENSEMBL baseMean log2FoldChange lfcSE  stat    pvalue      padj gene 
##    &lt;chr&gt;      &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;
##  1 ENSG00…    1209.          -4.67 0.178 -26.2 8.24e-152 1.37e-147 PGRM…
##  2 ENSG00…     277.           6.25 0.351  17.8 6.25e- 71 5.20e- 67 ZBED2
##  3 ENSG00…     630.           2.24 0.154  14.6 3.32e- 48 1.84e- 44 NUDT…
##  4 ENSG00…     220.           4.29 0.300  14.3 2.20e- 46 9.16e- 43 CST6 
##  5 ENSG00…    2692.           2.05 0.144  14.2 5.64e- 46 1.88e- 42 YWHAH
##  6 ENSG00…     381.           2.33 0.171  13.6 3.45e- 42 9.55e- 39 THG1L
##  7 ENSG00…     506.           2.53 0.191  13.2 8.45e- 40 2.01e- 36 EMP2 
##  8 ENSG00…     193.           3.44 0.271  12.7 7.18e- 37 1.49e- 33 STEA…
##  9 ENSG00…    1419.           3.00 0.240  12.5 6.39e- 36 1.18e- 32 ST3G…
## 10 ENSG00…    6134.           2.18 0.178  12.2 2.26e- 34 3.76e- 31 ITGB8
## # … with 13,709 more rows, and 1 more variable: ENTREZID &lt;chr&gt;</code></pre>
</div>
</div>
<div id="over-representation-analysis-ora" class="section level2">
<h2>
<span class="header-section-number">6.3</span> Over representation analysis (ORA)</h2>
<p>An over representation analysis relies on the hypergeometric distribution. The hypergeometric distribution is a discrete probability distribution that describes the probability of <span class="math inline">\(k\)</span> successes (random draws for which the object drawn has a specified feature) in <span class="math inline">\(n\)</span> draws, without replacement, from a finite population of size <span class="math inline">\(N\)</span> that contains exactly <span class="math inline">\(K\)</span> objects with that feature, wherein each draw is either a success or a failure.</p>
<p><span class="math display">\[ P(X = k) = \frac{\binom{N}{k} \binom{N-K}{n-k}}{\binom{N}{n}} \]</span></p>
<p>where</p>
<ul>
<li>
<span class="math inline">\(N\)</span> is the population size,</li>
<li>
<span class="math inline">\(K\)</span> is the number of success states in the population,</li>
<li>
<span class="math inline">\(n\)</span> is the number of draws (i.e. quantity drawn in each trial),</li>
<li>
<span class="math inline">\(k\)</span> is the number of observed successes,</li>
<li>
<span class="math inline">\(\binom{n}{k}\)</span> is a binomial coefficient <span class="math inline">\(\frac{n!}{k! (n-k)!}\)</span>.</li>
</ul>
<p>The example used to describe the distribution is an urn contain <span class="math inline">\(N\)</span> marbles of two colours. There are <span class="math inline">\(K\)</span> green and and <span class="math inline">\(N-K\)</span> red marbles in the urn. Drawing a green marble is defined as success, and a red marble failure. Using the formula above, we can compute the probability to draw <span class="math inline">\(k\)</span> green marbles from the urn.</p>
<p>The in frame of an enrichment analysis <span class="citation">(Rivals et al. <label for="tufte-mn-12" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-12" class="margin-toggle">2007<span class="marginnote">Rivals, I, L Personnaz, L Taing, and M C Potier. 2007. “Enrichment or Depletion of a Go Category Within a Class of Genes: Which Test?” <em>Bioinformatics</em> 23 (4): 401–7. doi:<a href="https://doi.org/10.1093/bioinformatics/btl633">10.1093/bioinformatics/btl633</a>.</span>)</span>, we use the following formulation to calculate a probabiliy that we have more green marbles than we would expect by change, i.e. there to be an over representation of green marbles.</p>
<p><span class="math display">\[ p = 1 - \sum_{i = 0}^{k - 1} \frac{\binom{N}{k} \binom{N-K}{n-k}}{\binom{N}{n}} \]</span></p>
<p>To perform an over representation analysis, we thus need to define:</p>
<ul>
<li>among all the genes (called the universe), which ones are differentially expressed (DE);</li>
<li>among all the genes, which ones are part of the gene set of interest.</li>
</ul>
<p>And fill out the following table and count the number of DE genes that are in the set of interest, the non-DE that are in the set, and the DE and non-DE genes that are not in the set:</p>
<table>
<thead><tr class="header">
<th align="left"></th>
<th align="left">GO</th>
<th align="left">not_GO</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">DE</td>
<td align="left">n</td>
<td align="left">p</td>
</tr>
<tr class="even">
<td align="left">not_DE</td>
<td align="left">m</td>
<td align="left">q</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## DE and GO
n &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>],
                      GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))

## not DE and GO
m &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.05</span>],
                      GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))

## DE and not GO
p &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">setdiff</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>],
                    GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))

## not DE not not GO
q &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">setdiff</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.05</span>],
                    GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cont_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(n, m, p, q), <span class="dt">nrow =</span> <span class="dv">2</span>)
<span class="kw">rownames</span>(cont_mat) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"DE"</span>, <span class="st">"not_DE"</span>)
<span class="kw">colnames</span>(cont_mat) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"GO"</span>, <span class="st">"not_GO"</span>)
cont_mat</code></pre></div>
<pre><code>##         GO not_GO
## DE     158   4026
## not_DE 201   9334</code></pre>
<p>We can now apply a Fisher's exact (or hypergeometric test) that will test whether we can identify a statistically enrichment of DE genes in the GO category.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fisher.test</span>(cont_mat, <span class="dt">alternative =</span> <span class="st">"greater"</span>)</code></pre></div>
<pre><code>## 
##  Fisher's Exact Test for Count Data
## 
## data:  cont_mat
## p-value = 3.119e-08
## alternative hypothesis: true odds ratio is greater than 1
## 95 percent confidence interval:
##  1.516306      Inf
## sample estimates:
## odds ratio 
##   1.822262</code></pre>
<p>Note that we could keep the default <code>alternative = "two.sided"</code> to test to over or under representation.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the ORA analysis above the <a href="GO:0005813" class="uri">GO:0005813</a> term.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>This approach is straightfoward and very fast. Its major drawback however is that we need to define a cutoff to differentiate DE from non-DE genes. Setting this threshold might have a effect on the results.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Try setting different DE genes and check if, in the cases above, this has and effect on the GO terms of interest.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2>
<span class="header-section-number">6.4</span> Gene set enrichment analysis (GSEA)</h2>
<p>Gene set enrichment analysis refers to a broad family of tests. Here, we will define the principles based on <span class="citation">(Subramanian et al. <label for="tufte-mn-13" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-13" class="margin-toggle">2005<span class="marginnote">Subramanian, A, P Tamayo, V K Mootha, S Mukherjee, B L Ebert, M A Gillette, A Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proc Natl Acad Sci U S A</em> 102 (43): 15545–50. doi:<a href="https://doi.org/10.1073/pnas.0506580102">10.1073/pnas.0506580102</a>.</span>)</span>, keeping in mind that the exact implementation will differ in different tools.</p>
<p>The major advantage of GSEA approaches is that they don't rely on defining DE genes. The first step is to <strong>order</strong> the genes of interest based on the statistics used; here, we will use the p-values. This is already the case for our <code>res_tbl</code> table. We also need to know which genes are in our set of interest.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_tbl<span class="op">$</span>inGO &lt;-<span class="st"> </span>res_tbl<span class="op">$</span>ENTREZID <span class="op">%in%</span><span class="st"> </span>GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID
dplyr<span class="op">::</span><span class="kw">select</span>(res_tbl, ENTREZID, padj, inGO)</code></pre></div>
<pre><code>## # A tibble: 13,719 x 3
##    ENTREZID      padj inGO 
##    &lt;chr&gt;        &lt;dbl&gt; &lt;lgl&gt;
##  1 10857    1.37e-147 FALSE
##  2 79413    5.20e- 67 FALSE
##  3 55270    1.84e- 44 FALSE
##  4 1474     9.16e- 43 FALSE
##  5 7533     1.88e- 42 FALSE
##  6 54974    9.55e- 39 FALSE
##  7 2013     2.01e- 36 FALSE
##  8 55240    1.49e- 33 FALSE
##  9 6482     1.18e- 32 FALSE
## 10 3696     3.76e- 31 TRUE 
## # … with 13,709 more rows</code></pre>
<p>We are now going to compute a score by traversing to ordered gene list and count a positive score when we encounter a gene in the gene set, and a -1 when the gene is not in the gene set. The positive score, computed by <code>set_ratio()</code> below, is defined by <span class="math inline">\(\frac{n_{genes} - n_{genes~in~set}}{n_{genes~in~set}}\)</span> so that the sum of all genes in the set and those not in the set becomes zero.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">set_ratio &lt;-<span class="st"> </span><span class="cf">function</span>(tbl) {
    (<span class="kw">nrow</span>(tbl) <span class="op">-</span><span class="st"> </span><span class="kw">sum</span>(tbl<span class="op">$</span>inGO))<span class="op">/</span><span class="kw">sum</span>(tbl<span class="op">$</span>inGO)
}</code></pre></div>
<p>The cumulative sum of these scores along the ordered gene list becomes the GSEA path which is plotted below, and the maximun score obtained along this path is the GSEA score.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gsea_path &lt;-<span class="st"> </span><span class="kw">ifelse</span>(res_tbl<span class="op">$</span>inGO, <span class="kw">set_ratio</span>(res_tbl), <span class="dv">-1</span>)
gsea_path &lt;-<span class="st"> </span><span class="kw">cumsum</span>(gsea_path)
gsea_score &lt;-<span class="st"> </span><span class="kw">max</span>(gsea_path)
<span class="kw">plot</span>(gsea_path, <span class="dt">type =</span> <span class="st">"l"</span>,
     <span class="dt">xlab =</span> <span class="st">"Ordered genes (padj)"</span>,
     <span class="dt">ylab =</span> <span class="st">"score"</span>,
     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">"GSEA score: "</span>, <span class="kw">round</span>(gsea_score)))
<span class="kw">abline</span>(<span class="dt">h =</span> gsea_score, <span class="dt">col =</span> <span class="st">"steelblue"</span>, <span class="dt">lty =</span> <span class="st">"dotted"</span>)</code></pre></div>
<div class="figure">
<span id="fig:gsea1"></span>
<p class="caption marginnote shownote">
Figure 6.3: Gene set enrichment analysis path for term <a href="GO:0005925" class="uri">GO:0005925</a>.
</p>
<img src="WSBIM2122_files/figure-html/gsea1-1.png" alt="Gene set enrichment analysis path for term GO:0005925." width="672">
</div>
<p>To be able to compute a statistical significance, we need to compute a null distribution of GSEA scores, i.e. a distribution of scores reflecting the absence of any enrichment. This is done by permuting the samples in the original data, repeating the statistical analysis, reorder the genes accodring to the new statistics (we used the adjusted p-value above) and compute a new GSEA score.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds_tmp &lt;-<span class="st"> </span>dds
dds_tmp<span class="op">$</span>Condition &lt;-<span class="st"> </span>dds_tmp<span class="op">$</span>Condition[<span class="kw">sample</span>(<span class="kw">ncol</span>(dds_tmp))]
<span class="kw">colData</span>(dds)</code></pre></div>
<pre><code>## DataFrame with 6 rows and 4 columns
##                Cell        Type Condition sizeFactor
##         &lt;character&gt; &lt;character&gt;  &lt;factor&gt;  &lt;numeric&gt;
## sample1       Cell1  Epithelial      mock   0.726883
## sample2       Cell1  Epithelial      mock   1.308664
## sample3       Cell1  Epithelial      mock   1.077368
## sample4       Cell1  Epithelial      KD     0.966025
## sample5       Cell1  Epithelial      KD     1.042111
## sample6       Cell1  Epithelial      KD     1.012664</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colData</span>(dds_tmp)</code></pre></div>
<pre><code>## DataFrame with 6 rows and 4 columns
##                Cell        Type Condition sizeFactor
##         &lt;character&gt; &lt;character&gt;  &lt;factor&gt;  &lt;numeric&gt;
## sample1       Cell1  Epithelial      mock   0.726883
## sample2       Cell1  Epithelial      KD     1.308664
## sample3       Cell1  Epithelial      KD     1.077368
## sample4       Cell1  Epithelial      KD     0.966025
## sample5       Cell1  Epithelial      mock   1.042111
## sample6       Cell1  Epithelial      mock   1.012664</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_tmp &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds_tmp) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">results</span>(<span class="dt">name =</span> <span class="st">"Condition_KD_vs_mock"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as_tibble</span>(<span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">left_join</span>(ensembl_to_geneName) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">inGO =</span> ENTREZID <span class="op">%in%</span><span class="st"> </span>GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ENTREZID),
           <span class="op">!</span><span class="kw">is.na</span>(padj),
           <span class="op">!</span><span class="kw">duplicated</span>(ENTREZID)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(ENSEMBL, ENTREZID, padj, inGO) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">arrange</span>(padj)

res_tmp &lt;-<span class="st"> </span>res_tmp <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">ifelse</span>(inGO, <span class="kw">set_ratio</span>(res_tmp), <span class="dv">-1</span>)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cumsum</span>(score))

res_tmp</code></pre></div>
<pre><code>## # A tibble: 19,093 x 5
##    ENSEMBL          ENTREZID  padj inGO  score
##    &lt;chr&gt;               &lt;int&gt; &lt;dbl&gt; &lt;lgl&gt; &lt;dbl&gt;
##  1 ENSG00000223972     84771 0.999 FALSE    -1
##  2 ENSG00000223972    727856 0.999 FALSE    -2
##  3 ENSG00000223972 100287102 0.999 FALSE    -3
##  4 ENSG00000223972 100287596 0.999 FALSE    -4
##  5 ENSG00000223972 102725121 0.999 FALSE    -5
##  6 ENSG00000227232    653635 0.999 FALSE    -6
##  7 ENSG00000278267 102466751 0.999 FALSE    -7
##  8 ENSG00000279457 102723897 0.999 FALSE    -8
##  9 ENSG00000230021 101928626 0.999 FALSE    -9
## 10 ENSG00000237491 105378580 0.999 FALSE   -10
## # … with 19,083 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">max</span>(res_tmp<span class="op">$</span>score)</code></pre></div>
<pre><code>## [1] 433.4621</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(res_tmp<span class="op">$</span>score, <span class="dt">type =</span> <span class="st">"l"</span>)</code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-119-1.png" width="672"></p>
<p>This approach is very time consuming, given that the statistical tests for all the genes need to be recomputer at each permutation. In addition, for experiments with limited number of samples, the number of permutations would be limited.</p>
<p>We start by defining a function that will repeat the steps above and return a list containing the GSEA score and path.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##' A function that assigns the permuted Conditions,
##' runs DESeq and a GSEA analysis
##'
##' @param x a DESeq object.
##' @param perm `character()` of length `ncol(x)` indicating the
##'     permutation to be tested.
##'
##' @return `numeric()` containing the GSEA path.
gsea_perm &lt;-<span class="st"> </span><span class="cf">function</span>(x, perm) {
    ## Set the Condition based on the permutation
    x<span class="op">$</span>Condition[perm] &lt;-<span class="st"> "mock"</span>
    x<span class="op">$</span>Condition[<span class="op">-</span>perm] &lt;-<span class="st"> "KD"</span>
    ## Run DESeq2, extract and annotate results
    <span class="kw">suppressMessages</span>(
        tbl &lt;-<span class="st"> </span><span class="kw">DESeq</span>(x) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">results</span>(<span class="dt">name =</span> <span class="st">"Condition_KD_vs_mock"</span>) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">as_tibble</span>(<span class="dt">rownames =</span> <span class="st">"ENSEMBL"</span>) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">left_join</span>(ensembl_to_geneName) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">mutate</span>(<span class="dt">inGO =</span> ENTREZID <span class="op">%in%</span><span class="st"> </span>GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ENTREZID),
                   <span class="op">!</span><span class="kw">is.na</span>(padj),
                   <span class="op">!</span><span class="kw">duplicated</span>(ENTREZID)) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">select</span>(ENSEMBL, ENTREZID, padj, inGO) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">arrange</span>(padj)
    )
    ## Compute GSEA results
    tbl &lt;-<span class="st"> </span>tbl <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">if_else</span>(inGO, <span class="kw">set_ratio</span>(tbl), <span class="dv">-1</span>)) <span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">mutate</span>(<span class="dt">score =</span> <span class="kw">cumsum</span>(score))
    <span class="kw">return</span>(tbl<span class="op">$</span>score)
}</code></pre></div>
<p>We will expecute this function on all unique permutations, corresponding on the 10 first columns (including the actual design, in the first column). Note that columns 11 to 20 are simply the opposite of the first 10.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(perms &lt;-<span class="st"> </span><span class="kw">combn</span>(<span class="dv">6</span>, <span class="dv">3</span>))</code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13] [,14]
## [1,]    1    1    1    1    1    1    1    1    1     1     2     2     2     2
## [2,]    2    2    2    2    3    3    3    4    4     5     3     3     3     4
## [3,]    3    4    5    6    4    5    6    5    6     6     4     5     6     5
##      [,15] [,16] [,17] [,18] [,19] [,20]
## [1,]     2     2     3     3     3     4
## [2,]     4     5     4     4     5     5
## [3,]     6     6     5     6     6     6</code></pre>
<p>We not apply our function on every permutation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">paths &lt;-<span class="st"> </span><span class="kw">apply</span>(perms[, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>], <span class="dv">2</span>, <span class="cf">function</span>(p) <span class="kw">gsea_perm</span>(dds, p))</code></pre></div>
<p>And calculate the scores</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scores &lt;-<span class="st"> </span><span class="kw">sapply</span>(paths, max)</code></pre></div>
<p>If we compare the actual score to the permutation scores, we see that it is the largest one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">density</span>(scores))
<span class="kw">rug</span>(scores)
<span class="kw">abline</span>(<span class="dt">v =</span> scores[<span class="dv">1</span>])</code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-123-1.png" width="672"></p>
<p>An empirical p-value can be computer by dividing the number of null scores that are greater than the real score divided by the number of permutations (the number of null scores). In our case, given that no null scores are greater, the nominal p-value would be 0.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(scores <span class="op">&gt;</span><span class="st"> </span>scores[<span class="dv">1</span>])<span class="op">/</span><span class="kw">length</span>(scores)</code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Below, we illustrate all GSEA path and confirm that the actual score is indeed the largest one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(paths[[<span class="dv">1</span>]], <span class="dt">type =</span> <span class="st">"l"</span>,
     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="kw">min</span>(<span class="kw">unlist</span>(paths)), <span class="kw">max</span>(scores)),
     <span class="dt">col =</span> <span class="st">"steelblue"</span>)
<span class="kw">grid</span>()
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">length</span>(paths))
    <span class="kw">lines</span>(paths[[i]], <span class="dt">type =</span> <span class="st">"l"</span>, <span class="dt">col =</span> <span class="st">"#00000060"</span>)</code></pre></div>
<div class="figure">
<span id="fig:unnamed-chunk-125"></span>
<p class="caption marginnote shownote">
Figure 6.4: Representation of all GSEA paths: real path (blue) and random paths (dotted grey).
</p>
<img src="WSBIM2122_files/figure-html/unnamed-chunk-125-1.png" alt="Representation of all GSEA paths: real path (blue) and random paths (dotted grey)." width="672">
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Given that we have 10 permutations, how many null scores greater that the actual GSEA score do we need for our results to become non significant (assuming we set an alpha of 0.05)?</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-17" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-17', 'sol-start-17')"></span>
</p>
<div id="sol-body-17" class="solution-body" style="display: none;">
<p>1 permutation score or more is enough to loose significativite, as a single value greater than 2244 produces a p-value of 1/10 = 0.1.</p>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>There exist various approaches to the GSEA analysis, that apply different permutation approaches to increate the number of possible permutations and reduce the running time.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the GSEA analysis above the <a href="GO:0005813" class="uri">GO:0005813</a> term.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>While there is not need to define a list of differentially expressed genes for the GSEA analysis, the genes need to be ordered, which can be done in different ways and lead to different results.</p>
</div>
<div id="using-the-clusterprofiler-package" class="section level2">
<h2>
<span class="header-section-number">6.5</span> Using the <code>clusterProfiler</code> package</h2>
<p>I practice, the analyses presented above are executed using any of <a href="http://bioconductor.org/packages/release/BiocViews.html#___GeneSetEnrichment">the very many</a> packages that are available. Here, we will demonstrate <em><a href="https://bioconductor.org/packages/3.11/clusterProfiler">clusterProfiler</a></em><label for="tufte-sn-14" class="margin-toggle sidenote-number">14</label><input type="checkbox" id="tufte-sn-14" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">14</span> <a href="https://yulab-smu.top/clusterProfiler-book/" class="uri">https://yulab-smu.top/clusterProfiler-book/</a></span>, that itself relies on other packages to perform the GSEA-related computations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"clusterProfiler"</span>)</code></pre></div>
<p>These dedicated packages will perform either of the two tests (or variations thereof) to all sets, adjust the computed p-values for multiplicity, and provide additional details and visualisations for further exploration of the results.</p>
<div id="ora-using-go-sets" class="section level3">
<h3>
<span class="header-section-number">6.5.1</span> ORA using GO sets</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">de_genes &lt;-<span class="st"> </span>res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>]

go_ora &lt;-<span class="st"> </span><span class="kw">enrichGO</span>(<span class="dt">gene =</span> de_genes,
                   <span class="dt">universe =</span> res_tbl<span class="op">$</span>ENTREZID,
                   <span class="dt">OrgDb =</span> org.Hs.eg.db,
                   <span class="dt">ont =</span> <span class="st">"CC"</span>,
                   <span class="dt">readable =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble
go_ora</code></pre></div>
<pre><code>## # A tibble: 17 x 9
##    ID     Description  GeneRatio BgRatio   pvalue p.adjust  qvalue geneID  Count
##    &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;int&gt;
##  1 GO:00… cytosolic r… 63/3819   100/11… 1.41e-10  9.67e-8 8.99e-8 RPL27/…    63
##  2 GO:00… cytosolic l… 40/3819   58/119… 7.22e- 9  2.47e-6 2.30e-6 RPL27/…    40
##  3 GO:00… polysomal r… 24/3819   32/119… 6.76e- 7  1.03e-4 9.59e-5 RPL38/…    24
##  4 GO:00… focal adhes… 158/3819  359/11… 7.03e- 7  1.03e-4 9.59e-5 ITGB8/…   158
##  5 GO:00… cell-substr… 159/3819  362/11… 7.53e- 7  1.03e-4 9.59e-5 ITGB8/…   159
##  6 GO:00… ribosomal s… 86/3819   180/11… 5.60e- 6  6.39e-4 5.94e-4 DAP3/M…    86
##  7 GO:00… membrane ra… 95/3819   210/11… 3.10e- 5  2.65e-3 2.47e-3 EMP2/S…    95
##  8 GO:00… membrane mi… 95/3819   210/11… 3.10e- 5  2.65e-3 2.47e-3 EMP2/S…    95
##  9 GO:00… cell surface 170/3819  415/11… 4.67e- 5  3.56e-3 3.31e-3 EMP2/I…   170
## 10 GO:00… membrane re… 97/3819   218/11… 5.70e- 5  3.91e-3 3.63e-3 EMP2/S…    97
## 11 GO:00… ribosome     96/3819   217/11… 8.12e- 5  5.06e-3 4.70e-3 DAP3/M…    96
## 12 GO:00… cytosolic s… 24/3819   39/119… 1.30e- 4  7.41e-3 6.89e-3 RPS24/…    24
## 13 GO:00… neuron to n… 101/3819  235/11… 2.06e- 4  1.08e-2 1.01e-2 NETO2/…   101
## 14 GO:00… postsynapti… 97/3819   226/11… 2.88e- 4  1.41e-2 1.31e-2 NETO2/…    97
## 15 GO:00… postsynapti… 93/3819   218/11… 4.87e- 4  2.22e-2 2.07e-2 NETO2/…    93
## 16 GO:00… large ribos… 54/3819   116/11… 6.50e- 4  2.78e-2 2.59e-2 MRPL10…    54
## 17 GO:00… asymmetric … 93/3819   220/11… 7.01e- 4  2.82e-2 2.62e-2 NETO2/…    93</code></pre>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the ORA analysis above using all GO namespaces. See <code>?enrichGO</code> for details.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="gsea-using-go-sets" class="section level3">
<h3>
<span class="header-section-number">6.5.2</span> GSEA using GO sets</h3>
<p>The functions that perform GSEA in <code>clusterProfiler</code> require the genes to be ordered in decreasing order. This indicates that the p-values can't be used without transformations. One could also use the log2 fold-change or the (absolute value) test statistics.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ordered_genes &lt;-<span class="st"> </span><span class="kw">abs</span>(res_tbl<span class="op">$</span>stat)
<span class="kw">names</span>(ordered_genes) &lt;-<span class="st"> </span>res_tbl<span class="op">$</span>ENTREZID
ordered_genes &lt;-<span class="st"> </span><span class="kw">sort</span>(ordered_genes, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)

go_gsea &lt;-<span class="st"> </span><span class="kw">gseGO</span>(<span class="dt">gene =</span> ordered_genes,
                 <span class="dt">OrgDb =</span> org.Hs.eg.db,
                 <span class="dt">scoreType =</span> <span class="st">"pos"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble</code></pre></div>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.62% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">go_gsea</code></pre></div>
<pre><code>## # A tibble: 503 x 11
##    ID    Description setSize enrichmentScore   NES  pvalue p.adjust qvalues
##    &lt;chr&gt; &lt;chr&gt;         &lt;int&gt;           &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
##  1 GO:0… epithelial…     405           0.471  1.38 2.61e-9  1.41e-5 1.09e-5
##  2 GO:0… positive r…     495           0.448  1.32 7.29e-8  1.64e-4 1.27e-4
##  3 GO:0… negative r…     445           0.451  1.32 1.09e-7  1.64e-4 1.27e-4
##  4 GO:0… positive r…     491           0.447  1.31 1.21e-7  1.64e-4 1.27e-4
##  5 GO:0… negative r…     452           0.448  1.32 1.55e-7  1.67e-4 1.30e-4
##  6 GO:0… actin fila…     295           0.471  1.37 2.14e-7  1.93e-4 1.50e-4
##  7 GO:0… actin cyto…     490           0.442  1.30 2.90e-7  2.24e-4 1.74e-4
##  8 GO:0… cardiovasc…     492           0.442  1.30 4.47e-7  3.02e-4 2.35e-4
##  9 GO:0… response t…     425           0.447  1.31 5.41e-7  3.25e-4 2.52e-4
## 10 GO:0… blood vess…     460           0.444  1.30 7.02e-7  3.80e-4 2.95e-4
## # … with 493 more rows, and 3 more variables: rank &lt;int&gt;, leading_edge &lt;chr&gt;,
## #   core_enrichment &lt;chr&gt;</code></pre>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the GSEA analysis above using all GO namespaces. See <code>?enrichGO</code> for details.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Compare and discuss the use of these different ways to order the genes for the GSEA analysis.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Compare the GO and GSEA results. Which GO terms are shared or unique to each approach?</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-18" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-18', 'sol-start-18')"></span>
</p>
<div id="sol-body-18" class="solution-body" style="display: none;">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">go_cc &lt;-<span class="st"> </span><span class="kw">full_join</span>(go_ora <span class="op">%&gt;%</span>
<span class="st">                   </span><span class="kw">filter</span>(p.adjust <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span>
<span class="st">                   </span>dplyr<span class="op">::</span><span class="kw">select</span>(ID, p.adjust),
                 go_gsea <span class="op">%&gt;%</span>
<span class="st">                   </span><span class="kw">filter</span>(p.adjust <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">%&gt;%</span>
<span class="st">                   </span>dplyr<span class="op">::</span><span class="kw">select</span>(ID, p.adjust),
                 <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">"ID"</span> =<span class="st"> "ID"</span>))

## shared
<span class="kw">na.omit</span>(go_cc)</code></pre></div>
<pre><code>## # A tibble: 0 x 3
## # … with 3 variables: ID &lt;chr&gt;, p.adjust.x &lt;dbl&gt;, p.adjust.y &lt;dbl&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## ORA only
<span class="kw">filter</span>(go_cc, <span class="kw">is.na</span>(p.adjust.y))</code></pre></div>
<pre><code>## # A tibble: 12 x 3
##    ID           p.adjust.x p.adjust.y
##    &lt;chr&gt;             &lt;dbl&gt;      &lt;dbl&gt;
##  1 GO:0022626 0.0000000967         NA
##  2 GO:0022625 0.00000247           NA
##  3 GO:0042788 0.000103             NA
##  4 GO:0005925 0.000103             NA
##  5 GO:0030055 0.000103             NA
##  6 GO:0044391 0.000639             NA
##  7 GO:0045121 0.00265              NA
##  8 GO:0098857 0.00265              NA
##  9 GO:0009986 0.00356              NA
## 10 GO:0098589 0.00391              NA
## 11 GO:0005840 0.00506              NA
## 12 GO:0022627 0.00741              NA</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## GSEA only
<span class="kw">filter</span>(go_cc, <span class="kw">is.na</span>(p.adjust.x))</code></pre></div>
<pre><code>## # A tibble: 170 x 3
##    ID         p.adjust.x p.adjust.y
##    &lt;chr&gt;           &lt;dbl&gt;      &lt;dbl&gt;
##  1 GO:0030855         NA  0.0000141
##  2 GO:0043068         NA  0.000164 
##  3 GO:0045596         NA  0.000164 
##  4 GO:0043065         NA  0.000164 
##  5 GO:0008285         NA  0.000167 
##  6 GO:0007015         NA  0.000193 
##  7 GO:0030036         NA  0.000224 
##  8 GO:0072358         NA  0.000302 
##  9 GO:0009611         NA  0.000325 
## 10 GO:0001568         NA  0.000380 
## # … with 160 more rows</code></pre>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="using-kegg-sets" class="section level3">
<h3>
<span class="header-section-number">6.5.3</span> Using KEGG sets</h3>
<p>The <code>enrichKEGG()</code> and <code>gseKEGG()</code> functions can be used to perform the ORA and GSEA analyses againt the KEGG sets.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Perform ORA and GSEA analysis using the KEGG set.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="using-msigdb-sets" class="section level3">
<h3>
<span class="header-section-number">6.5.4</span> Using MSigDb sets</h3>
<p>We will use the hallmark gene set as an example here, but any set described above can be utilised.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"msigdbr"</span>)
msig_h &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">"Homo sapiens"</span>, <span class="dt">category =</span> <span class="st">"H"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">ont =</span> gs_name, <span class="dt">gene =</span> entrez_gene)
msig_h</code></pre></div>
<pre><code>## # A tibble: 7,321 x 2
##    ont                    gene
##    &lt;chr&gt;                 &lt;int&gt;
##  1 HALLMARK_ADIPOGENESIS    19
##  2 HALLMARK_ADIPOGENESIS 11194
##  3 HALLMARK_ADIPOGENESIS 10449
##  4 HALLMARK_ADIPOGENESIS    33
##  5 HALLMARK_ADIPOGENESIS    34
##  6 HALLMARK_ADIPOGENESIS    35
##  7 HALLMARK_ADIPOGENESIS    47
##  8 HALLMARK_ADIPOGENESIS    50
##  9 HALLMARK_ADIPOGENESIS    51
## 10 HALLMARK_ADIPOGENESIS   112
## # … with 7,311 more rows</code></pre>
<p>The <code>msig_h</code> table can now be used with the <code>enricher()</code> and <code>GSEA()</code> functions from the <code>clusterProfiler</code> package to perform ORA and GSEA analyses on the MSigDB hallmark gene set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msig_ora &lt;-<span class="st"> </span><span class="kw">enricher</span>(<span class="dt">gene =</span> de_genes,
                     <span class="dt">universe =</span> res_tbl<span class="op">$</span>ENTREZID,
                     <span class="dt">TERM2GENE =</span> msig_h) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble
msig_ora</code></pre></div>
<pre><code>## # A tibble: 3 x 9
##   ID      Description GeneRatio BgRatio  pvalue p.adjust  qvalue geneID    Count
##   &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;     &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;     &lt;int&gt;
## 1 HALLMA… HALLMARK_H… 82/1211   164/33… 1.32e-4  0.00661 0.00584 10370/94…    82
## 2 HALLMA… HALLMARK_M… 34/1211   58/3353 3.58e-4  0.00896 0.00792 1844/231…    34
## 3 HALLMA… HALLMARK_A… 53/1211   106/33… 2.04e-3  0.0339  0.0300  3566/16/…    53</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msig_gsea &lt;-<span class="st"> </span><span class="kw">GSEA</span>(ordered_genes, <span class="dt">TERM2GENE =</span> msig_h, <span class="dt">scoreType =</span> <span class="st">"pos"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>as_tibble</code></pre></div>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.62% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msig_gsea</code></pre></div>
<pre><code>## # A tibble: 17 x 11
##    ID    Description setSize enrichmentScore   NES  pvalue p.adjust qvalues
##    &lt;chr&gt; &lt;chr&gt;         &lt;int&gt;           &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
##  1 HALL… HALLMARK_E…     129           0.545  1.54 2.77e-7  1.39e-5 7.59e-6
##  2 HALL… HALLMARK_H…     164           0.503  1.44 7.99e-6  2.00e-4 1.09e-4
##  3 HALL… HALLMARK_M…      58           0.600  1.62 1.72e-5  2.86e-4 1.57e-4
##  4 HALL… HALLMARK_P…     177           0.485  1.39 4.13e-5  5.16e-4 2.82e-4
##  5 HALL… HALLMARK_G…     175           0.471  1.35 1.78e-4  1.78e-3 9.74e-4
##  6 HALL… HALLMARK_T…     161           0.475  1.36 2.35e-4  1.96e-3 1.07e-3
##  7 HALL… HALLMARK_U…     122           0.486  1.37 3.58e-4  2.56e-3 1.40e-3
##  8 HALL… HALLMARK_K…     124           0.481  1.36 5.54e-4  3.46e-3 1.90e-3
##  9 HALL… HALLMARK_X…     134           0.472  1.34 8.47e-4  4.71e-3 2.58e-3
## 10 HALL… HALLMARK_E…     152           0.465  1.33 1.04e-3  5.20e-3 2.85e-3
## 11 HALL… HALLMARK_A…     106           0.485  1.36 1.32e-3  5.98e-3 3.27e-3
## 12 HALL… HALLMARK_A…     124           0.463  1.30 2.87e-3  1.19e-2 6.54e-3
## 13 HALL… HALLMARK_M…     116           0.460  1.30 4.05e-3  1.56e-2 8.53e-3
## 14 HALL… HALLMARK_T…      50           0.526  1.41 4.95e-3  1.77e-2 9.67e-3
## 15 HALL… HALLMARK_I…     113           0.463  1.30 6.13e-3  2.04e-2 1.12e-2
## 16 HALL… HALLMARK_I…     147           0.440  1.25 6.88e-3  2.15e-2 1.18e-2
## 17 HALL… HALLMARK_E…     159           0.430  1.23 8.44e-3  2.48e-2 1.36e-2
## # … with 3 more variables: rank &lt;int&gt;, leading_edge &lt;chr&gt;,
## #   core_enrichment &lt;chr&gt;</code></pre>
</div>
</div>
<div id="visualisation-of-enrichment-analyses" class="section level2">
<h2>
<span class="header-section-number">6.6</span> Visualisation of enrichment analyses</h2>
<p>The <code>clusterProfiler</code> documentation provides a chapter on the <a href="http://yulab-smu.top/clusterProfiler-book/chapter12.html">visualization of functional enrichment results</a>.</p>
<p>Another useful visualisation, that links the enrichment results back to the whole set of results is to highlight the genes in a particular set of interest on the volcano plot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sel &lt;-<span class="st"> </span>res_tbl<span class="op">$</span>ENTREZID <span class="op">%in%</span><span class="st"> </span>GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID
<span class="kw">plot</span>(res_tbl<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw">log10</span>(res_tbl<span class="op">$</span>padj))
<span class="kw">points</span>(res_tbl<span class="op">$</span>log2FoldChange[sel],
     <span class="op">-</span><span class="kw">log10</span>(res_tbl<span class="op">$</span>padj)[sel],
     <span class="dt">col =</span> <span class="st">"red"</span>)            
<span class="kw">grid</span>()</code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-127-1.png" width="672"></p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Extract the Entrez ids from the top hit in the <code>msig_gsea</code> results above (available in <code>core_enrichment</code>) and visualise the results on a volcano plot.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-19" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-19', 'sol-start-19')"></span>
</p>
<div id="sol-body-19" class="solution-body" style="display: none;">
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">entrez_ids &lt;-<span class="st"> </span><span class="kw">strsplit</span>(msig_gsea<span class="op">$</span>core_enrichment[[<span class="dv">1</span>]], <span class="st">"/"</span>)[[<span class="dv">1</span>]]

res_tbl <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">inGO =</span> ENTREZID <span class="op">%in%</span><span class="st"> </span>entrez_ids) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">arrange</span>(inGO) <span class="op">%&gt;%</span><span class="st"> </span>## to plot inGO points last (on top)
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> log2FoldChange,
               <span class="dt">y =</span> <span class="op">-</span><span class="kw">log10</span>(padj),
               <span class="dt">colour =</span> inGO)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-128-1.png" width="672"></p>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="discussion" class="section level2">
<h2>
<span class="header-section-number">6.7</span> Discussion</h2>
<p>Enrichment analyses are powerful techniques, as they allow to integrate thousands of univariate results into biologically driven sets.</p>
<p>Their interpretation isn't however always straightforward when different methods (ORA and GSEA) and implementations (different GSEA permutation strategies) or user-set parameters (DE cutoffs in ORA, ordering criteria in GSEA) provide different results.</p>

</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-rnaseq.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-ms.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2020-10-29
</p>
</div>
</div>



</body>
</html>
