<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 11 Session information | Omics Data Analysis" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2020-12-12" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_CHTML-full" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 11 Session information | Omics Data Analysis</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.16/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.9.2.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.52.2/plotly-latest.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a href="sec-gsea.html"><span class="toc-section-number">6</span> Enrichment analyses</a>
<a href="sec-ms.html"><span class="toc-section-number">7</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">8</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">9</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">10</span> Annex</a>
<a id="active-page" href="sec-si.html"><span class="toc-section-number">11</span> Session information</a><ul class="toc-sections">
<li class="toc"><a href="#sec:setup2"> R package setup</a></li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec:si" class="section level1">
<h1>
<span class="header-section-number">Chapter 11</span> Session information</h1>
<p>The following packages have been used to generate this document.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 4.0.0 (2020-04-24)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.5 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/libf77blas.so.3.10.3
## LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] plotly_4.9.2.1              factoextra_1.0.7           
##  [3] gplots_3.1.0                limma_3.46.0               
##  [5] QFeatures_1.0.0             MultiAssayExperiment_1.16.0
##  [7] PSM_0.2.0                   magrittr_2.0.1             
##  [9] Spectra_1.1.6               ProtGenerics_1.23.5        
## [11] BiocParallel_1.24.1         msigdbr_7.2.1              
## [13] clusterProfiler_3.18.0      GO.db_3.12.1               
## [15] org.Hs.eg.db_3.12.0         AnnotationDbi_1.52.0       
## [17] biomaRt_2.46.0              patchwork_1.1.0            
## [19] DESeq2_1.30.0               SummarizedExperiment_1.20.0
## [21] Biobase_2.50.0              MatrixGenerics_1.2.0       
## [23] matrixStats_0.57.0          GenomicRanges_1.42.0       
## [25] GenomeInfoDb_1.26.1         IRanges_2.24.0             
## [27] S4Vectors_0.28.0            BiocGenerics_0.36.0        
## [29] DT_0.16                     forcats_0.5.0              
## [31] stringr_1.4.0               dplyr_1.0.2                
## [33] purrr_0.3.4                 readr_1.4.0                
## [35] tidyr_1.1.2                 tibble_3.0.4               
## [37] tidyverse_1.3.0             ggplot2_3.3.2              
## [39] rWSBIM2122_0.3.0           
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.1.4              tidyselect_1.1.0        RSQLite_2.2.1          
##   [4] htmlwidgets_1.5.2       grid_4.0.0              scatterpie_0.1.5       
##   [7] munsell_0.5.0           codetools_0.2-16        preprocessCore_1.52.0  
##  [10] msdata_0.30.0           withr_2.3.0             colorspace_2.0-0       
##  [13] GOSemSim_2.16.1         highr_0.8               knitr_1.30             
##  [16] rstudioapi_0.13         ggsignif_0.6.0          DOSE_3.16.0            
##  [19] labeling_0.4.2          GenomeInfoDbData_1.2.4  polyclip_1.10-0        
##  [22] bit64_4.0.5             farver_2.0.3            downloader_0.4         
##  [25] vctrs_0.3.5             generics_0.1.0          xfun_0.19              
##  [28] BiocFileCache_1.14.0    R6_2.5.0                graphlayouts_0.7.1     
##  [31] locfit_1.5-9.4          MsCoreUtils_1.2.0       AnnotationFilter_1.14.0
##  [34] bitops_1.0-6            fgsea_1.16.0            DelayedArray_0.16.0    
##  [37] assertthat_0.2.1        scales_1.1.1            ggraph_2.0.4           
##  [40] enrichplot_1.10.1       gtable_0.3.0            Cairo_1.5-12.2         
##  [43] tidygraph_1.2.0         rlang_0.4.9             genefilter_1.72.0      
##  [46] mzR_2.24.1              splines_4.0.0           rstatix_0.6.0          
##  [49] lazyeval_0.2.2          broom_0.7.2             abind_1.4-5            
##  [52] BiocManager_1.30.10     yaml_2.2.1              reshape2_1.4.4         
##  [55] modelr_0.1.8            crosstalk_1.1.0.1       backports_1.2.0        
##  [58] qvalue_2.22.0           tools_4.0.0             bookdown_0.21          
##  [61] ellipsis_0.3.1          RColorBrewer_1.1-2      Rcpp_1.0.5             
##  [64] plyr_1.8.6              progress_1.2.2          zlibbioc_1.36.0        
##  [67] RCurl_1.98-1.2          ps_1.4.0                prettyunits_1.1.1      
##  [70] ggpubr_0.4.0            openssl_1.4.3           viridis_0.5.1          
##  [73] cowplot_1.1.0           haven_2.3.1             ggrepel_0.8.2          
##  [76] fs_1.5.0                data.table_1.13.2       openxlsx_4.2.3         
##  [79] DO.db_2.9               reprex_0.3.0            msmbstyle_0.0.11       
##  [82] hms_0.5.3               evaluate_0.14           xtable_1.8-4           
##  [85] XML_3.99-0.5            rio_0.5.16              readxl_1.3.1           
##  [88] gridExtra_2.3           compiler_4.0.0          KernSmooth_2.23-18     
##  [91] ncdf4_1.17              crayon_1.3.4            shadowtext_0.0.7       
##  [94] htmltools_0.5.0         geneplotter_1.68.0      lubridate_1.7.9.2      
##  [97] DBI_1.1.0               tweenr_1.0.1            dbplyr_2.0.0           
## [100] MASS_7.3-53             rappdirs_0.3.1          BiocStyle_2.18.1       
## [103] car_3.0-10              Matrix_1.2-18           cli_2.2.0              
## [106] igraph_1.2.6            pkgconfig_2.0.3         rvcheck_0.1.8          
## [109] foreign_0.8-80          xml2_1.3.2              annotate_1.68.0        
## [112] XVector_0.30.0          rvest_0.3.6             digest_0.6.27          
## [115] rmarkdown_2.5           cellranger_1.1.0        fastmatch_1.1-0        
## [118] curl_4.3                gtools_3.8.2            lifecycle_0.2.0        
## [121] jsonlite_1.7.1          carData_3.0-4           viridisLite_0.3.0      
## [124] askpass_1.1             fansi_0.4.1             pillar_1.4.7           
## [127] lattice_0.20-41         httr_1.4.2              survival_3.2-7         
## [130] glue_1.4.2              zip_2.1.1               png_0.1-7              
## [133] bit_4.0.4               ggforce_0.3.2           stringi_1.5.3          
## [136] blob_1.2.1              caTools_1.18.0          memoise_1.1.0</code></pre>
<div id="sec:setup2" class="section level2">
<h2>
<span class="header-section-number">11.1</span> R package setup</h2>
<p>To install all necessary packages to run all the code, please execute the following code:</p>
<pre><code>pkgs &lt;- c("AnnotationDbi", "base", "Biobase", "BiocGenerics",
          "BiocParallel", "biomaRt", "clusterProfiler", "datasets", "DESeq2",
          "dplyr", "DT", "factoextra", "forcats", "GenomeInfoDb",
          "GenomicRanges", "ggplot2", "GO.db", "gplots", "graphics", "grDevices",
          "IRanges", "limma", "magrittr", "MatrixGenerics", "matrixStats",
          "methods", "msigdbr", "MultiAssayExperiment", "org.Hs.eg.db",
          "parallel", "patchwork", "plotly", "ProtGenerics", "PSM", "purrr",
          "QFeatures", "readr", "rWSBIM2122", "S4Vectors", "Spectra", "stats",
          "stats4", "stringr", "SummarizedExperiment", "tibble", "tidyr",
          "tidyverse", "utils")</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw">require</span>(<span class="st">"BiocManager"</span>))
   <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(pkgs)</code></pre></div>

</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-anx.html"><button class="btn btn-default">Previous</button></a>
</p>
<p class="build-date">Page built: 
2020-12-12
</p>
</div>
</div>



</body>
</html>
