<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 11 Session information | Omics Data Analysis" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2020-10-09" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 11 Session information | Omics Data Analysis</title>

<script src="libs/header-attrs-2.4/header-attrs.js"></script>
<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a href="sec-gsea.html"><span class="toc-section-number">6</span> Enrichment analyses</a>
<a href="sec-ms.html"><span class="toc-section-number">7</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">8</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">9</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">10</span> Annex</a>
<a id="active-page" href="sec-si.html"><span class="toc-section-number">11</span> Session information</a><ul class="toc-sections">
<li class="toc"><a href="#sec:setup2"> R package setup</a></li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec:si" class="section level1" number="11">
<h1>
<span class="header-section-number">Chapter 11</span> Session information</h1>
<p>The following packages have been used to generate this document.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="sec-si.html#cb316-1" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 Patched (2020-07-19 r78897)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Manjaro Linux
## 
## Matrix products: default
## BLAS:   /usr/lib/libblas.so.3.9.0
## LAPACK: /usr/lib/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] msigdbr_7.2.1               clusterProfiler_3.16.1     
##  [3] GO.db_3.11.4                org.Hs.eg.db_3.11.4        
##  [5] AnnotationDbi_1.50.3        biomaRt_2.44.1             
##  [7] patchwork_1.0.1             DESeq2_1.28.1              
##  [9] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
## [11] matrixStats_0.57.0          Biobase_2.48.0             
## [13] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
## [15] IRanges_2.22.2              S4Vectors_0.26.1           
## [17] BiocGenerics_0.34.0         DT_0.15                    
## [19] forcats_0.5.0               stringr_1.4.0              
## [21] dplyr_1.0.2                 purrr_0.3.4                
## [23] readr_1.4.0                 tidyr_1.1.2                
## [25] tibble_3.0.3                tidyverse_1.3.0            
## [27] ggplot2_3.3.2               rWSBIM2122_0.3.0           
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1         backports_1.1.10     fastmatch_1.1-0     
##   [4] BiocFileCache_1.12.1 plyr_1.8.6           igraph_1.2.6        
##   [7] splines_4.0.2        BiocParallel_1.22.0  crosstalk_1.1.0.1   
##  [10] urltools_1.7.3       digest_0.6.25        msmbstyle_0.0.11    
##  [13] htmltools_0.5.0      GOSemSim_2.14.2      viridis_0.5.1       
##  [16] fansi_0.4.1          magrittr_1.5         memoise_1.1.0       
##  [19] graphlayouts_0.7.0   annotate_1.66.0      modelr_0.1.8        
##  [22] askpass_1.1          enrichplot_1.8.1     prettyunits_1.1.1   
##  [25] colorspace_1.4-1     blob_1.2.1           rvest_0.3.6         
##  [28] rappdirs_0.3.1       ggrepel_0.8.2        haven_2.3.1         
##  [31] xfun_0.18            crayon_1.3.4         RCurl_1.98-1.2      
##  [34] jsonlite_1.7.1       scatterpie_0.1.5     genefilter_1.70.0   
##  [37] survival_3.2-3       glue_1.4.2           polyclip_1.10-0     
##  [40] gtable_0.3.0         zlibbioc_1.34.0      XVector_0.28.0      
##  [43] scales_1.1.1         DOSE_3.15.2          DBI_1.1.0           
##  [46] Rcpp_1.0.5           viridisLite_0.3.0    xtable_1.8-4        
##  [49] progress_1.2.2       gridGraphics_0.5-0   europepmc_0.4       
##  [52] bit_4.0.4            htmlwidgets_1.5.2    httr_1.4.2          
##  [55] fgsea_1.14.0         RColorBrewer_1.1-2   ellipsis_0.3.1      
##  [58] pkgconfig_2.0.3      XML_3.99-0.5         farver_2.0.3        
##  [61] dbplyr_1.4.4         locfit_1.5-9.4       utf8_1.1.4          
##  [64] ggplotify_0.0.5      tidyselect_1.1.0     labeling_0.3        
##  [67] rlang_0.4.8          reshape2_1.4.4       munsell_0.5.0       
##  [70] cellranger_1.1.0     tools_4.0.2          downloader_0.4      
##  [73] cli_2.0.2            generics_0.0.2       RSQLite_2.2.1       
##  [76] ggridges_0.5.2       broom_0.7.1          evaluate_0.14       
##  [79] yaml_2.2.1           knitr_1.30           bit64_4.0.5         
##  [82] fs_1.5.0             tidygraph_1.2.0      ggraph_2.0.3        
##  [85] DO.db_2.9            xml2_1.3.2           BiocStyle_2.16.1    
##  [88] compiler_4.0.2       rstudioapi_0.11      curl_4.3            
##  [91] reprex_0.3.0         tweenr_1.0.1         geneplotter_1.66.0  
##  [94] stringi_1.5.3        highr_0.8            lattice_0.20-41     
##  [97] Matrix_1.2-18        vctrs_0.3.4          pillar_1.4.6        
## [100] lifecycle_0.2.0     
##  [ reached getOption("max.print") -- omitted 20 entries ]</code></pre>
<div id="sec:setup2" class="section level2" number="11.1">
<h2>
<span class="header-section-number">11.1</span> R package setup</h2>
<p>To install all necessary packages to run all the code, please execute
the following code:</p>
<pre><code>pkgs &lt;- c("AnnotationDbi", "base", "Biobase", "BiocGenerics",
          "biomaRt", "clusterProfiler", "datasets", "DelayedArray", "DESeq2",
          "dplyr", "DT", "forcats", "GenomeInfoDb", "GenomicRanges", "ggplot2",
          "GO.db", "graphics", "grDevices", "IRanges", "matrixStats", "methods",
          "msigdbr", "org.Hs.eg.db", "parallel", "patchwork", "purrr", "readr",
          "rWSBIM2122", "S4Vectors", "stats", "stats4", "stringr",
          "SummarizedExperiment", "tibble", "tidyr", "tidyverse", "utils")</code></pre>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="sec-si.html#cb319-1" aria-hidden="true"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">require</span>(<span class="st">"BiocManager"</span>))</span>
<span id="cb319-2"><a href="sec-si.html#cb319-2" aria-hidden="true"></a>   <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb319-3"><a href="sec-si.html#cb319-3" aria-hidden="true"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(pkgs)</span></code></pre></div>

</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-anx.html"><button class="btn btn-default">Previous</button></a>
</p>
<p class="build-date">Page built: 
2020-10-09
</p>
</div>
</div>



</body>
</html>
