<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 6 Enrichment analyses | Omics Data Analysis" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2020-10-03" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS_CHTML-full" type="text/javascript"></script>

<meta name="description" content="Course material for the Omics Data Analysis (WSBIM2122) course at UCLouvain.">

<title>Chapter 6 Enrichment analyses | Omics Data Analysis</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }

code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Omics Data Analysis<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html">Preamble</a>
<a href="sec-cli.html"><span class="toc-section-number">1</span> The shell</a>
<a href="sec-rmd.html"><span class="toc-section-number">2</span> R markdown</a>
<a href="sec-linmod.html"><span class="toc-section-number">3</span> Linear models</a>
<a href="sec-hts.html"><span class="toc-section-number">4</span> High throughput sequencing</a>
<a href="sec-rnaseq.html"><span class="toc-section-number">5</span> Differential expression analysis</a>
<a id="active-page" href="sec-gsea.html"><span class="toc-section-number">6</span> Enrichment analyses</a><ul class="toc-sections">
<li class="toc"><a href="#introduction"> Introduction</a></li>
<li class="toc"><a href="#gene-sets"> Gene sets</a></li>
<li class="toc"><a href="#statistical-analyses"> Statistical analyses</a></li>
<li class="toc"><a href="#using-the-clusterprofiler-package"> Using the <code>clusterProfiler</code> package</a></li>
</ul>
<a href="sec-ms.html"><span class="toc-section-number">7</span> Mass spectrometry</a>
<a href="sec-prot.html"><span class="toc-section-number">8</span> Quantitative proteomics data analysis</a>
<a href="sec-ccl.html"><span class="toc-section-number">9</span> Conclusions</a>
<a href="sec-anx.html"><span class="toc-section-number">10</span> Annex</a>
<a href="sec-si.html"><span class="toc-section-number">11</span> Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
<html><body><div id="sec:gsea" class="section level1">
<h1>
<span class="header-section-number">Chapter 6</span> Enrichment analyses</h1>
<p><strong>Motivation</strong>: at the end of this chapter, the students will be able
to address the following points:</p>
<ul>
<li>What are the next steps after a differential expression analysis?</li>
<li>Why are enrichment analyses useful?</li>
<li>Understanding over-enrichment analyses.</li>
<li>Understanding a gene set enrichment analysis.</li>
<li>Application of the <em><a href="https://bioconductor.org/packages/3.11/clusterProfiler">clusterProfiler</a></em> package.</li>
</ul>
<div id="introduction" class="section level2">
<h2>
<span class="header-section-number">6.1</span> Introduction</h2>
<ul>
<li><p>Differential expression analysis is <strong>univariate</strong> - each gene is
tested on its own. This probably doesn’t reflect the underlying
biology - genes work in conjunction, not in isolation. One wouldn’t
expect that the effect of a drug, or a mutation, … would lead to
the perturbation of a single gene expression.</p></li>
<li><p>The univariate approach expects (or tests for) significant changes
in single genes. Moderate effects in many (related) genes cannot, by
definition be identified as statistically significant, even it such
an effect may be biologically more plausible that one of a few large
ones.</p></li>
<li><p>The goal of an enrichment analysis is to test for a group of related
genes, called <strong>gene sets</strong>, and test whether the genes within these
sets are enriched for differentially expression.</p></li>
</ul>
</div>
<div id="gene-sets" class="section level2">
<h2>
<span class="header-section-number">6.2</span> Gene sets</h2>
<p>While nothing would stop a user to create their own gene sets, the
sets that are generally used stem from community-maintained resources.</p>
<div id="gene-ontology-go" class="section level3">
<h3>
<span class="header-section-number">6.2.1</span> Gene Ontology (GO)</h3>
<p>The <a href="http://geneontology.org/">Gene Ontology</a> <span class="citation">(Ashburner et al. <label for="tufte-mn-11" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-11" class="margin-toggle">2000<span class="marginnote">Ashburner, M, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, et al. 2000. “Gene Ontology: Tool for the Unification of Biology. The Gene Ontology Consortium.” <em>Nat Genet</em> 25 (1): 25–29. <a href="https://doi.org/10.1038/75556" class="uri">https://doi.org/10.1038/75556</a>.</span>)</span>
defines GO terms. These terms are based on a controlled vocabulary (to
resuse the same words in a consistent way) and relations<label for="tufte-sn-11" class="margin-toggle sidenote-number">11</label><input type="checkbox" id="tufte-sn-11" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">11</span> Example of relations between terms are, for example, <em>is_a</em>
(mitosis <em>is_a</em> cell cycle phase) or <em>part_of</em> (mitosis <em>part_of</em>
M phase of mitotic cell cycle).</span> that
define the links between terms.</p>
<p>These terms are classed into three categories, called <em>namespaces</em>:</p>
<ul>
<li>Molecular Function (MF): molecular activities of gene products</li>
<li>Cellular Component (CC): where gene products are active</li>
<li>Biological Process (BP): pathways and larger processes made up of
the activities of multiple gene products</li>
</ul>
<p>Here’s an example of GO term <code>GO:0007155</code>, that describes cell adhesion.</p>
<pre><code>A Term from the GO ontology: GO:0007155 
 Label: cell adhesion
  The attachment of a cell, either to another cell or to an underlying
  substrate such as the extracellular matrix, via cell adhesion
  molecules.</code></pre>
<p>If you look at the GO term <code>GO:0007155</code> <a href="http://amigo.geneontology.org/amigo/term/GO:0007155">entry on the Gene Ontology
page</a>, you can
find more details and, if you scroll down, example genes that are
annotated with that term.</p>
<div class="figure" style="text-align: center">
<span id="fig:unnamed-chunk-109"></span>
<p class="caption marginnote shownote">
Figure 6.1: Gene ontology entry for the GO term for cell adhesion.
</p>
<img src="figs/go-screenshot.png" alt="Gene ontology entry for the GO term for cell adhesion." width="100%">
</div>
<p>The whole Gene Ontology is can be accessed in R with the
<em><a href="https://bioconductor.org/packages/3.11/GO.db">GO.db</a></em> package.</p>
<p>In the code chunk below, we query the <code>GO.db</code> package through the
<code>org.Hs.eg.db</code> interface. This <em>org.db</em> for Homo sapiens enables to
perform various queries for human genes, such as retrieving all gene
symbols and ENTREZ identifiers (the <code>columns</code> below) that are
annotated with a GO term (the <code>keys</code> below) of interest.</p>
<p>The GO term of interest here is focal adhesion:</p>
<pre><code>A cell-substrate junction that anchors the cell to the extracellular
matrix and that forms a point of termination of actin filaments. In
insects focal adhesion has also been referred to as hemi-adherens
junction (HAJ).</code></pre>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb257-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"GO.db"</span>)</a>
<a class="sourceLine" id="cb257-2" data-line-number="2"><span class="kw">library</span>(<span class="st">"org.Hs.eg.db"</span>)</a>
<a class="sourceLine" id="cb257-3" data-line-number="3">GO_<span class="dv">0005925</span> &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db,</a>
<a class="sourceLine" id="cb257-4" data-line-number="4">                     <span class="dt">keys =</span> <span class="st">"GO:0005925"</span>,</a>
<a class="sourceLine" id="cb257-5" data-line-number="5">                     <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>),</a>
<a class="sourceLine" id="cb257-6" data-line-number="6">                     <span class="dt">keytype =</span> <span class="st">"GO"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb257-7" data-line-number="7"><span class="st">    </span>as_tibble <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb257-8" data-line-number="8"><span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(ENTREZID))</a></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb259-1" data-line-number="1">GO_<span class="dv">0005925</span></a></code></pre></div>
<pre><code>## # A tibble: 405 x 5
##    GO         EVIDENCE ONTOLOGY ENTREZID SYMBOL
##    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
##  1 GO:0005925 HDA      CC       60       ACTB  
##  2 GO:0005925 HDA      CC       70       ACTC1 
##  3 GO:0005925 ISS      CC       71       ACTG1 
##  4 GO:0005925 HDA      CC       81       ACTN4 
##  5 GO:0005925 HDA      CC       87       ACTN1 
##  6 GO:0005925 IMP      CC       88       ACTN2 
##  7 GO:0005925 IMP      CC       89       ACTN3 
##  8 GO:0005925 HDA      CC       102      ADAM10
##  9 GO:0005925 HDA      CC       118      ADD1  
## 10 GO:0005925 HDA      CC       214      ALCAM 
## # … with 395 more rows</code></pre>
<p>We have 405 genes matching this GO term. There are
thus 405 genes in the GO_0005925 GO set.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the code above to extract the genes annotated with the
<code>GO:0005813</code> term for the centrosome:</p>
<pre><code>A structure comprised of a core structure (in most organisms, a pair
of centrioles) and peripheral material from which a
microtubule-based structure, such as a spindle apparatus, is
organized. Centrosomes occur close to the nucleus during interphase
in many eukaryotic cells, though in animal cells it changes
continually during the cell-division cycle.</code></pre>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-16" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-16', 'sol-start-16')"></span>
</p>
<div id="sol-body-16" class="solution-body" style="display: none;">
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb262-1" data-line-number="1">GO_<span class="dv">0005813</span> &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db,</a>
<a class="sourceLine" id="cb262-2" data-line-number="2">                     <span class="dt">keys =</span> <span class="st">"GO:0005813"</span>,</a>
<a class="sourceLine" id="cb262-3" data-line-number="3">                     <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>),</a>
<a class="sourceLine" id="cb262-4" data-line-number="4">                     <span class="dt">keytype =</span> <span class="st">"GO"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb262-5" data-line-number="5"><span class="st">    </span>as_tibble <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb262-6" data-line-number="6"><span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(ENTREZID))</a></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb264-1" data-line-number="1">GO_<span class="dv">0005813</span></a></code></pre></div>
<pre><code>## # A tibble: 483 x 5
##    GO         EVIDENCE ONTOLOGY ENTREZID SYMBOL
##    &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt; 
##  1 GO:0005813 IDA      CC       324      APC   
##  2 GO:0005813 IDA      CC       328      APEX1 
##  3 GO:0005813 IDA      CC       402      ARL2  
##  4 GO:0005813 IDA      CC       403      ARL3  
##  5 GO:0005813 IDA      CC       468      ATF4  
##  6 GO:0005813 IDA      CC       576      ADGRB2
##  7 GO:0005813 IBA      CC       582      BBS1  
##  8 GO:0005813 IDA      CC       585      BBS4  
##  9 GO:0005813 IDA      CC       598      BCL2L1
## 10 GO:0005813 IMP      CC       636      BICD1 
## # … with 473 more rows</code></pre>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="kyoto-encyclopedia-of-genes-and-genomes-kegg" class="section level3">
<h3>
<span class="header-section-number">6.2.2</span> Kyoto Encyclopedia of Genes and Genomes (KEGG)</h3>
<p><a href="https://www.genome.jp/kegg/pathway.html">KEGG</a> pathway is a
collection of manually drawn and curated pathway maps representing
current knowledge of the molecular interaction, reaction and relation
networks.</p>
<p>The figure below shows the <a href="https://www.genome.jp/kegg-bin/show_pathway?hsa04110">pathways for the cell
cycle</a> in
humans.</p>
<div class="figure" style="text-align: center">
<span id="fig:unnamed-chunk-112"></span>
<p class="caption marginnote shownote">
Figure 6.2: KEGG pathway for cell cycle.
</p>
<img src="figs/hsa04110.png" alt="KEGG pathway for cell cycle." width="100%">
</div>
<p>The <em><a href="https://bioconductor.org/packages/3.11/KEGGREST">KEGGREST</a></em> package provides a client interface to the KEGG server.</p>
</div>
<div id="reactome" class="section level3">
<h3>
<span class="header-section-number">6.2.3</span> Reactome</h3>
<p>Alike KEGG patway, <a href="https://reactome.org/">Reactome</a> is a free,
open-source, curated and peer-reviewed pathway database. The
Bioconductor <em><a href="https://bioconductor.org/packages/3.11/reactome.db">reactome.db</a></em> package provides
access to reactome maps and annotations within R.</p>
</div>
<div id="molecular-signatures-database-mysigdb" class="section level3">
<h3>
<span class="header-section-number">6.2.4</span> Molecular Signatures Database (MySigDB)</h3>
<p>MSigDB is a collection of annotated gene sets for use with GSEA
software. The MSigDB gene sets are divided into 9 collections:</p>
<ul>
<li>Hallmark gene sets (H) are coherently expressed signatures derived
by aggregating many MSigDB gene sets to represent well-defined
biological states or processes.</li>
<li>Positional gene sets (C1) for each human chromosome and cytogenetic band.</li>
<li>Curated gene sets (C2) from online pathway databases, publications
in PubMed, and knowledge of domain experts.</li>
<li>Regulatory target gene sets (C3) based on gene target predictions
for microRNA seed sequences and predicted transcription factor
binding sites.</li>
<li>Computational gene sets (C4) defined by mining large collections of
cancer-oriented microarray data.</li>
<li>Ontology gene sets (C5) consist of genes annotated by the same
ontology term.</li>
<li>Oncogenic signature gene (C6) sets defined directly from microarray
gene expression data from cancer gene perturbations.</li>
<li>Immunologic signature gene sets (C7) defined directly from
microarray gene expression data from immunologic studies.</li>
<li>Cell type signature gene sets (C8) curated from cluster markers
identified in single-cell sequencing studies of human tissue.</li>
</ul>
</div>
</div>
<div id="statistical-analyses" class="section level2">
<h2>
<span class="header-section-number">6.3</span> Statistical analyses</h2>
<p>To illustrate enrichment analyses, we will use the DESeq2 results
stored in the the <code>res_tbl</code> variable, computed in the previous
chapter.</p>
<p>We will focus on the genes that have an adjusted p-value (those that
have been tested) and that have unique ENTREZ gene identifiers.</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb266-1" data-line-number="1">res_tbl &lt;-<span class="st"> </span>res_tbl <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb266-2" data-line-number="2"><span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(ENTREZID),</a>
<a class="sourceLine" id="cb266-3" data-line-number="3">           <span class="op">!</span><span class="kw">is.na</span>(padj),</a>
<a class="sourceLine" id="cb266-4" data-line-number="4">           <span class="op">!</span><span class="kw">duplicated</span>(res_tbl<span class="op">$</span>ENTREZID))</a>
<a class="sourceLine" id="cb266-5" data-line-number="5"></a>
<a class="sourceLine" id="cb266-6" data-line-number="6">res_tbl<span class="op">$</span>ENTREZID &lt;-<span class="st"> </span><span class="kw">as.character</span>(res_tbl<span class="op">$</span>ENTREZID)</a>
<a class="sourceLine" id="cb266-7" data-line-number="7">res_tbl</a></code></pre></div>
<pre><code>## # A tibble: 13,719 x 9
##    ENSEMBL baseMean log2FoldChange lfcSE  stat    pvalue      padj gene 
##    &lt;chr&gt;      &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;
##  1 ENSG00…    1209.          -4.67 0.178 -26.2 8.24e-152 1.37e-147 PGRM…
##  2 ENSG00…     277.           6.25 0.351  17.8 6.25e- 71 5.20e- 67 ZBED2
##  3 ENSG00…     630.           2.24 0.154  14.6 3.32e- 48 1.84e- 44 NUDT…
##  4 ENSG00…     220.           4.29 0.300  14.3 2.20e- 46 9.16e- 43 CST6 
##  5 ENSG00…    2692.           2.05 0.144  14.2 5.64e- 46 1.88e- 42 YWHAH
##  6 ENSG00…     381.           2.33 0.171  13.6 3.45e- 42 9.55e- 39 THG1L
##  7 ENSG00…     506.           2.53 0.191  13.2 8.45e- 40 2.01e- 36 EMP2 
##  8 ENSG00…     193.           3.44 0.271  12.7 7.18e- 37 1.49e- 33 STEA…
##  9 ENSG00…    1419.           3.00 0.240  12.5 6.39e- 36 1.18e- 32 ST3G…
## 10 ENSG00…    6134.           2.18 0.178  12.2 2.26e- 34 3.76e- 31 ITGB8
## # … with 13,709 more rows, and 1 more variable: ENTREZID &lt;chr&gt;</code></pre>
<div id="over-enrichment-analysis-ora" class="section level3">
<h3>
<span class="header-section-number">6.3.1</span> Over-enrichment analysis (ORA)</h3>
<p>To perform an over-enrichment analysis, we need to define:</p>
<ul>
<li>among all the genes, which ones are differentially expressed (DE);</li>
<li>among all the genes, which ones are part of the gene set of
interest.</li>
</ul>
<p>And fill out the following table and count the number of DE genes that
are in the set of interest, the non-DE that are in the set, and the DE
and non-DE genes that are not in the set:</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb268-1" data-line-number="1">     <span class="op">|</span><span class="st"> </span>GO <span class="op">|</span><span class="st"> </span><span class="op">!</span>GO</a>
<a class="sourceLine" id="cb268-2" data-line-number="2"><span class="op">-----+----+------</span></a>
<a class="sourceLine" id="cb268-3" data-line-number="3"><span class="st">  </span>DE <span class="op">|</span><span class="st"> </span>n  <span class="op">|</span><span class="st">  </span>p</a>
<a class="sourceLine" id="cb268-4" data-line-number="4"> <span class="op">!</span>DE <span class="op">|</span><span class="st"> </span>m  <span class="op">|</span><span class="st">  </span>q</a></code></pre></div>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb269-1" data-line-number="1"><span class="co">## DE and GO</span></a>
<a class="sourceLine" id="cb269-2" data-line-number="2">n &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>],</a>
<a class="sourceLine" id="cb269-3" data-line-number="3">                      GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))</a>
<a class="sourceLine" id="cb269-4" data-line-number="4"></a>
<a class="sourceLine" id="cb269-5" data-line-number="5"></a>
<a class="sourceLine" id="cb269-6" data-line-number="6"><span class="co">## !DE and GO</span></a>
<a class="sourceLine" id="cb269-7" data-line-number="7">m &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.05</span>],</a>
<a class="sourceLine" id="cb269-8" data-line-number="8">                      GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))</a>
<a class="sourceLine" id="cb269-9" data-line-number="9"></a>
<a class="sourceLine" id="cb269-10" data-line-number="10"><span class="co">## DE and !GO</span></a>
<a class="sourceLine" id="cb269-11" data-line-number="11">p &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">setdiff</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>],</a>
<a class="sourceLine" id="cb269-12" data-line-number="12">                    GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))</a>
<a class="sourceLine" id="cb269-13" data-line-number="13"></a>
<a class="sourceLine" id="cb269-14" data-line-number="14"><span class="co">## !DE not !GO</span></a>
<a class="sourceLine" id="cb269-15" data-line-number="15">q &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">setdiff</span>(res_tbl<span class="op">$</span>ENTREZID[res_tbl<span class="op">$</span>padj <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.05</span>],</a>
<a class="sourceLine" id="cb269-16" data-line-number="16">                    GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID))</a></code></pre></div>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb270-1" data-line-number="1">cont_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(n, m, p, q), <span class="dt">nrow =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb270-2" data-line-number="2"><span class="kw">rownames</span>(cont_mat) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"DE"</span>, <span class="st">"!DE"</span>)</a>
<a class="sourceLine" id="cb270-3" data-line-number="3"><span class="kw">colnames</span>(cont_mat) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"GO"</span>, <span class="st">"!GO"</span>)</a>
<a class="sourceLine" id="cb270-4" data-line-number="4">cont_mat</a></code></pre></div>
<pre><code>##      GO  !GO
## DE  158 4026
## !DE 201 9334</code></pre>
<p>We can now apply a Fisher’s exact (or hypergeometric test) that will
test whether we can identify a statistically enrichment of DE genes in
the GO category.</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb272-1" data-line-number="1"><span class="kw">fisher.test</span>(cont_mat)</a></code></pre></div>
<pre><code>## 
## 	Fisher's Exact Test for Count Data
## 
## data:  cont_mat
## p-value = 5.285e-08
## alternative hypothesis: true odds ratio is not equal to 1
## 95 percent confidence interval:
##  1.465125 2.263592
## sample estimates:
## odds ratio 
##   1.822262</code></pre>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Repeat the analysis above the the <a href="GO:0005813" class="uri">GO:0005813</a> term.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>This approach is straightfoward and very fast. Its major drawback
however is that we need to define a cutoff to differentiate DE from
non-DE genes. Setting this threshold might have a effect on the
results.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Try setting different DE genes and check if, in the cases above, this
has and effect on the GO terms of interest.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level3">
<h3>
<span class="header-section-number">6.3.2</span> Gene set enrichment analysis (GSEA)</h3>
<p>Gene set enrichment analysis refers to a broad family of tests. Here,
we will define the principles based on <span class="citation">(Subramanian et al. <label for="tufte-mn-12" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-12" class="margin-toggle">2005<span class="marginnote">Subramanian, A, P Tamayo, V K Mootha, S Mukherjee, B L Ebert, M A Gillette, A Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proc Natl Acad Sci U S A</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102" class="uri">https://doi.org/10.1073/pnas.0506580102</a>.</span>)</span>, keeping in
mind that the exact implementation will differ in different tools.</p>
<p>The major advantage of GSEA approaches is that they don’t rely on
defining DE genes. The first step is to <strong>order</strong> the genes of
interest based on the statistics used, typically the p-values. This is
already the case for our <code>res_tbl</code> table. We also need to know which
genes are in our set of interest.</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb274-1" data-line-number="1">res_tbl<span class="op">$</span>inGO &lt;-<span class="st"> </span>res_tbl<span class="op">$</span>ENTREZID <span class="op">%in%</span><span class="st"> </span>GO_<span class="dv">0005925</span><span class="op">$</span>ENTREZID</a>
<a class="sourceLine" id="cb274-2" data-line-number="2">dplyr<span class="op">::</span><span class="kw">select</span>(res_tbl, ENTREZID, padj, inGO)</a></code></pre></div>
<pre><code>## # A tibble: 13,719 x 3
##    ENTREZID      padj inGO 
##    &lt;chr&gt;        &lt;dbl&gt; &lt;lgl&gt;
##  1 10857    1.37e-147 FALSE
##  2 79413    5.20e- 67 FALSE
##  3 55270    1.84e- 44 FALSE
##  4 1474     9.16e- 43 FALSE
##  5 7533     1.88e- 42 FALSE
##  6 54974    9.55e- 39 FALSE
##  7 2013     2.01e- 36 FALSE
##  8 55240    1.49e- 33 FALSE
##  9 6482     1.18e- 32 FALSE
## 10 3696     3.76e- 31 TRUE 
## # … with 13,709 more rows</code></pre>
<p>We are now going to compute a score by traversing to ordered gene list
and count a positive score when we encounter a gene in the gene set,
and a -1 when the gene is not in the gene set. The positive score,
called <code>set_ratio</code> below, is computeted as as <span class="math inline">\(\frac{n_{genes} - n_{genes~in~set}}{n_{genes~in~set}}\)</span> so that the sum of all genes in
the set and those not in the set becomes zero.</p>
<p>The cumulative sum of these scores along the ordered gene list becomes
the GSEA path which is plotted below, and the maximun score obtained
along this path is the GSEA score.</p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb276-1" data-line-number="1">set_ratio &lt;-<span class="st"> </span>(<span class="kw">nrow</span>(res_tbl) <span class="op">-</span><span class="st"> </span><span class="kw">sum</span>(res_tbl<span class="op">$</span>inGO))<span class="op">/</span><span class="kw">sum</span>(res_tbl<span class="op">$</span>inGO)</a>
<a class="sourceLine" id="cb276-2" data-line-number="2">gsea_path &lt;-<span class="st"> </span><span class="kw">ifelse</span>(res_tbl<span class="op">$</span>inGO, set_ratio, <span class="dv">-1</span>)</a>
<a class="sourceLine" id="cb276-3" data-line-number="3">gsea_path &lt;-<span class="st"> </span><span class="kw">cumsum</span>(gsea_path)</a>
<a class="sourceLine" id="cb276-4" data-line-number="4">gsea_score &lt;-<span class="st"> </span><span class="kw">max</span>(gsea_path)</a>
<a class="sourceLine" id="cb276-5" data-line-number="5"><span class="kw">plot</span>(gsea_path, <span class="dt">type =</span> <span class="st">"l"</span>,</a>
<a class="sourceLine" id="cb276-6" data-line-number="6">     <span class="dt">xlab =</span> <span class="st">"Ordered genes (padj)"</span>, </a>
<a class="sourceLine" id="cb276-7" data-line-number="7">     <span class="dt">ylab =</span> <span class="st">"score"</span>,</a>
<a class="sourceLine" id="cb276-8" data-line-number="8">     <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">"GSEA score: "</span>, <span class="kw">round</span>(gsea_score)))</a>
<a class="sourceLine" id="cb276-9" data-line-number="9"><span class="kw">abline</span>(<span class="dt">h =</span> gsea_score, <span class="dt">col =</span> <span class="st">"steelblue"</span>, <span class="dt">lty =</span> <span class="st">"dotted"</span>)</a></code></pre></div>
<div class="figure">
<span id="fig:unnamed-chunk-120"></span>
<p class="caption marginnote shownote">
Figure 6.3: Gene set enrichment analysis path for term <a href="GO:0005925" class="uri">GO:0005925</a>.
</p>
<img src="WSBIM2122_files/figure-html/unnamed-chunk-120-1.png" alt="Gene set enrichment analysis path for term GO:0005925." width="672">
</div>
<p>To be able to compute a statistical significance, we need to compute a
null distribution of GSEA scores, i.e. a distribution of scores
reflecting the absence of any enrichment. This would typically be
obtained by permuting the samples in the original data, repeating the
statistical analysis, reorder the genes accodring to the new
statistics (we used the adjusted p-value above) and compute a new GSEA
score. This would be repeated a certain number of times.</p>
<p>This is very time consuming the statistical tests for all the genes
need to be recomputer. In addition, for experiments with limited
number of samples, the number of permutations would be limited.</p>
<p>Below, we are simply going to generate 100 random permutations of the
gene list to simulate the procedure above and calculate the GSEA
score.</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb277-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">123</span>)</a>
<a class="sourceLine" id="cb277-2" data-line-number="2">gsea_null &lt;-<span class="st"> </span><span class="kw">replicate</span>(<span class="dv">100</span>, {</a>
<a class="sourceLine" id="cb277-3" data-line-number="3">    res_tmp &lt;-<span class="st"> </span>res_tbl</a>
<a class="sourceLine" id="cb277-4" data-line-number="4">    res_tmp<span class="op">$</span>padj &lt;-<span class="st"> </span><span class="kw">sample</span>(res_tmp<span class="op">$</span>padj)</a>
<a class="sourceLine" id="cb277-5" data-line-number="5">    res_tmp &lt;-<span class="st"> </span><span class="kw">arrange</span>(res_tmp, padj)</a>
<a class="sourceLine" id="cb277-6" data-line-number="6">    tmp_path &lt;-<span class="st"> </span><span class="kw">ifelse</span>(res_tmp<span class="op">$</span>inGO, set_ratio, <span class="dv">-1</span>)</a>
<a class="sourceLine" id="cb277-7" data-line-number="7">    tmp_path &lt;-<span class="st"> </span><span class="kw">cumsum</span>(tmp_path)</a>
<a class="sourceLine" id="cb277-8" data-line-number="8">    <span class="kw">max</span>(tmp_path)</a>
<a class="sourceLine" id="cb277-9" data-line-number="9">})</a>
<a class="sourceLine" id="cb277-10" data-line-number="10"></a>
<a class="sourceLine" id="cb277-11" data-line-number="11"><span class="kw">hist</span>(gsea_null)</a></code></pre></div>
<p><img src="WSBIM2122_files/figure-html/unnamed-chunk-121-1.png" width="672"></p>
<p>The best score we could get would correspond to a permutation starting
with all the 359 genes in the set:</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb278-1" data-line-number="1">gsea_max &lt;-<span class="st"> </span><span class="kw">sum</span>(res_tbl<span class="op">$</span>inGO) <span class="op">*</span><span class="st"> </span>set_ratio</a>
<a class="sourceLine" id="cb278-2" data-line-number="2">gsea_max</a></code></pre></div>
<pre><code>## [1] 13360</code></pre>
<p>while the worst score, where all the 359 genes are
at the end of the gene list would be zero.</p>
<p>All the possibilities mentioned above are shown on the plots below.</p>
<div class="figure fullwidth">
<span id="fig:unnamed-chunk-123"></span>
<img src="WSBIM2122_files/figure-html/unnamed-chunk-123-1.png" alt="Representation of all GSEA paths: real path (blue), best and worse possible paths (red) and random paths (dotted grey)." width="768"><p class="caption marginnote shownote">
Figure 6.4: Representation of all GSEA paths: real path (blue), best and worse possible paths (red) and random paths (dotted grey).
</p>
</div>
<p>An empirical p-value can be computer by dividing the number of null
scores that are greater than the real score divided by the number of
permutations (the number of null scores). In our case, given that no
null scores are greater, the nominal p-value would be 0.</p>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Given that we have 100 permutations, how many null scores greater that
the actual GSEA score do we need for our results to become non
significant (assuming we set an alpha of 0.05)?</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="solution">
<p class="solution-begin">
► Solution<span id="sol-start-17" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-17', 'sol-start-17')"></span>
</p>
<div id="sol-body-17" class="solution-body" style="display: none;">
<p>6 scores or more, as 6 values greater than 2249.3983287 produce a
p-values of 6/100 = 0.06.</p>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
</div>
</div>
<div id="using-the-clusterprofiler-package" class="section level2">
<h2>
<span class="header-section-number">6.4</span> Using the <code>clusterProfiler</code> package</h2>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb280-1" data-line-number="1"><span class="kw">library</span>(<span class="st">"clusterProfiler"</span>)</a>
<a class="sourceLine" id="cb280-2" data-line-number="2">ego &lt;-<span class="st"> </span><span class="kw">enrichGO</span>(<span class="dt">gene =</span> <span class="kw">as.character</span>(res_tbl<span class="op">$</span>ENTREZ[res_tbl<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>]),</a>
<a class="sourceLine" id="cb280-3" data-line-number="3">                <span class="dt">universe =</span> <span class="kw">as.character</span>(res_tbl<span class="op">$</span>ENTREZ),</a>
<a class="sourceLine" id="cb280-4" data-line-number="4">                <span class="dt">OrgDb =</span> org.Hs.eg.db,</a>
<a class="sourceLine" id="cb280-5" data-line-number="5">                <span class="dt">ont =</span> <span class="st">"CC"</span>,</a>
<a class="sourceLine" id="cb280-6" data-line-number="6">                <span class="dt">pAdjustMethod =</span> <span class="st">"BH"</span>,</a>
<a class="sourceLine" id="cb280-7" data-line-number="7">                <span class="dt">pvalueCutoff =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb280-8" data-line-number="8">                <span class="dt">qvalueCutoff =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb280-9" data-line-number="9">                <span class="dt">readable =</span> <span class="ot">TRUE</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb280-10" data-line-number="10"><span class="st">    </span>as_tibble</a>
<a class="sourceLine" id="cb280-11" data-line-number="11">ego</a></code></pre></div>

</div>
</div></body></html>

<p style="text-align: center;">
<a href="sec-rnaseq.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-ms.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2020-10-03
</p>
</div>
</div>



</body>
</html>
